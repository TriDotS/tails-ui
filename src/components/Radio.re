let component = ReasonReact.statelessComponent("Radio");

let make = (~checked, ~disabled, ~handleChange, ~label, _children) => {
  ...component,
  render: _self =>
    <div className="flex items-center">
      <input
        _type="radio"
        defaultChecked=(Js.Boolean.to_js_boolean(checked))
        disabled
        onChange=((_) => handleChange())
      />
      (
        label !== "" ?
          <label className="ml-2 text-sm">
            (ReasonReact.stringToElement(label))
          </label> :
          ReasonReact.nullElement
      )
    </div>
};

let default =
  ReasonReact.wrapReasonForJs(~component, jsProps =>
    make(
      ~checked=jsProps##checked,
      ~disabled=jsProps##disabled,
      ~handleChange=jsProps##handleChange,
      ~label=jsProps##label,
      [||]
    )
  );
