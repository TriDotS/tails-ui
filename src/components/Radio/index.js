import PropTypes from 'prop-types'
import React from 'react'

const Radio = ({ checked, disabled, handleChange, label, ...props }) => (
  <div className="flex items-center">
    <input
      type="radio"
      checked={checked}
      disabled={disabled}
      onChange={handleChange}
      {...props}
    />
    {label && <label className="ml-2 text-sm">{label}</label>}
  </div>
)

Radio.propTypes = {
  checked: PropTypes.bool,
  disabled: PropTypes.bool,
  handleChange: PropTypes.func,
  label: PropTypes.string
}

export default Radio
