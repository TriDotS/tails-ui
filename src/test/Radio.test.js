import Radio from '../components/Radio'
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(<Radio />, div)
})

test('radio snapshot', () => {
  const tree = renderer.create(<Radio />).toJSON()
  expect(tree).toMatchSnapshot()
})
