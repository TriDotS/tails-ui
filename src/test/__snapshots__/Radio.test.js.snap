// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`radio snapshot 1`] = `
<div
  className="flex items-center"
>
  <input
    checked={undefined}
    disabled={undefined}
    onChange={undefined}
    type="radio"
  />
</div>
`;
